integration/helpers/trustedclusters.go

```diff
@@ -124,18 +124,32 @@ func WaitForNodeCount(ctx context.Context, t *TeleInstance, clusterName string,
 		if err != nil {
 			return trace.Wrap(err)
 		}
+
+		// Validate that the site cache contains the expected count.
 		accessPoint, err := remoteSite.CachingAccessPoint()
 		if err != nil {
 			return trace.Wrap(err)
 		}
+
 		nodes, err := accessPoint.GetNodes(ctx, defaults.Namespace)
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		if len(nodes) == count {
-			return nil
+		if len(nodes) != count {
+			return trace.BadParameter("cache contained %v nodes, but wanted to find %v nodes", len(nodes), count)
 		}
-		return trace.BadParameter("found %v nodes, but wanted to find %v nodes", len(nodes), count)
+
+		// Validate that the site watcher contains the expected count.
+		watcher, err := remoteSite.NodeWatcher()
+		if err != nil {
+			return trace.Wrap(err)
+		}
```

Comment:
So without this it was possible to for a node to land into the access point's cache but not being processed yet by the node watcher?